### PR TITLE
EPMRPP-37278 || Method for dashboard removing has been updated, ACL permissions update

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/auth/permissions/Permissions.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ private Permissions() {
 
 	public static final String CAN_ADMINISTRATE_OBJECT = "hasPermission(returnObject, 'aclFullPermission') || hasRole('ADMINISTRATOR')";
 
-	public static final String CAN_READ_OBJECT = "hasPermission(returnObject, 'aclReadPermission')";
+	public static final String CAN_READ_OBJECT = "hasPermission(returnObject, 'aclReadPermission') || hasPermission(returnObject, 'aclFullPermission')";
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/auth/permissions/Permissions.java]: replace with "||" + CAN_ADMINISTRATE_OBJECT

File: src/main/java/com/epam/ta/reportportal/auth/permissions/Permissions.java [hunk 2]
```diff
@@ -44,7 +44,7 @@ private Permissions() {
...
 
 	public static final String CAN_READ_OBJECT_FILTER = "hasPermission(filterObject, 'aclReadPermission')";
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/auth/permissions/Permissions.java]: replace this with can read object

File: src/main/java/com/epam/ta/reportportal/core/dashboard/GetDashboardHandler.java [hunk 3]
```diff
@@ -32,14 +32,24 @@
...
-	 * Get dashboard by id
+	 * Get dashboard on which use have {@link com.epam.ta.reportportal.auth.permissions.AclReadPermission} by id
 	 *
 	 * @param dashboardId    Dashboard id
 	 * @param projectDetails Project details
 	 * @param user           User
 	 * @return dashboard
 	 */
-	Dashboard getDashboard(Long dashboardId, ProjectDetails projectDetails, ReportPortalUser user);
+	Dashboard getPermittedDashboard(Long dashboardId, ProjectDetails projectDetails, ReportPortalUser user);
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/dashboard/GetDashboardHandler.java]: let's remove a word Dashboard from here

File: src/main/java/com/epam/ta/reportportal/core/dashboard/GetDashboardHandler.java [hunk 4]
```diff
@@ -32,14 +32,24 @@
...
+
+	/**
+	 * Get dashboard on which use have {@link com.epam.ta.reportportal.auth.permissions.AclFullPermission} by id
+	 *
+	 * @param dashboardId    Dashboard id
+	 * @param projectDetails Project details
+	 * @param user           User
+	 * @return dashboard
+	 */
+	Dashboard getAdministratedDashboard(Long dashboardId, ProjectDetails projectDetails, ReportPortalUser user);
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/dashboard/GetDashboardHandler.java]: and here

File: src/main/java/com/epam/ta/reportportal/core/widget/GetWidgetHandler.java [hunk 5]
```diff
@@ -31,12 +31,20 @@
...
 
 	/**
-	 * Get widget by id with permission check
+	 * Get widget on which user have {@link com.epam.ta.reportportal.auth.permissions.AclReadPermission} by id
 	 *
 	 * @param widgetId Widget id
 	 * @return Allowed widget.
 	 */
-	Widget findById(Long widgetId);
+	Widget findPermittedById(Long widgetId);
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/widget/GetWidgetHandler.java]: let's give a name like in dashboard

File: src/main/java/com/epam/ta/reportportal/core/dashboard/GetDashboardHandler.java (hunk 3 above)
Comment (pbortnik) [reply src/main/java/com/epam/ta/reportportal/core/dashboard/GetDashboardHandler.java]: check if we really need projectDetails and user in params?

Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/dashboard/GetDashboardHandler.java]: removed

File: src/main/java/com/epam/ta/reportportal/auth/permissions/Permissions.java (hunk 2 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/auth/permissions/Permissions.java]: replaced

File: src/main/java/com/epam/ta/reportportal/auth/permissions/Permissions.java (hunk 1 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/auth/permissions/Permissions.java]: replaced

File: src/main/java/com/epam/ta/reportportal/core/dashboard/GetDashboardHandler.java (hunk 4 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/dashboard/GetDashboardHandler.java]: done

File: src/main/java/com/epam/ta/reportportal/core/widget/GetWidgetHandler.java (hunk 5 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/widget/GetWidgetHandler.java]: done

**Pull request merged**